src/town_gui.cpp

```diff
@@ -297,6 +316,16 @@ struct TownAuthorityWindow : Window {
 	{
 		this->SetDirty();
 	}
+
+	void OnInvalidateData(int data = 0, bool is_gui = true) override
```

Comment:
all other OnInvalidateData functions call the second parameter `gui_scope`. Should probably match...